tests/istio.mk

```diff
@@ -92,10 +92,21 @@ e2e_version_skew: istioctl generate_yaml
 e2e_all:
 	$(MAKE) --keep-going e2e_simple e2e_mixer e2e_bookinfo e2e_dashboard e2e_upgrade
 
-JUNIT_E2E_XML ?= $(ISTIO_OUT)/junit_e2e-all.xml
-e2e_all_junit_report: | $(JUNIT_REPORT)
+JUNIT_E2E_XML ?= $(ISTIO_OUT)/junit.xml
+TARGET ?= e2e_all
+with_junit_report: | $(JUNIT_REPORT)
```

Comment:
This recipe is correct and robust ever since its introduction on the e2e_all target. It guarantees junit xml generation regardless of test exit status, and always exit with the same status as that of the test without letting the parser masking nonzero status. The point is to get this right at one place and reuse everywhere else to reduce duplicated code (and hence duplicated bugs). Doing so has a higher chance of catching issue like [this PR](https://github.com/istio/istio/pull/4220).

Reply:
Right now we have something that works - and is easier to read, doesn't rely on arcane make and shell code. The goal was to keep make simple, so it doesn't turn into a bazel. 

I would prefer this PR just uploads the generated xml without rewriting unreleated code that work.